app/src/main/java/clarkson/ee408/tictactoev4/MainActivity.java

```diff
@@ -211,10 +244,20 @@ private class ButtonHandler implements View.OnClickListener {
         public void onClick(View v) {
             Log.d("button clicked", "button clicked");
 
-            for (int row = 0; row < TicTacToe.SIDE; row++)
-                for (int column = 0; column < TicTacToe.SIDE; column++)
-                    if (v == buttons[row][column])
+            for (int row = 0; row < TicTacToe.SIDE; row++) {
+                for (int column = 0; column < TicTacToe.SIDE; column++) {
+                    if (v == buttons[row][column]) {
+                        // Calculate move index
+                        int move = row * TicTacToe.SIDE + column;
+
+                        // 1️⃣ Send move to server first
```

Comment:
Generally, don't use unicode characters in your code unless you need to print a unicode character
```suggestion
                        // 1 - Send move to server first
```